itest/godot/TestRunner.gd

```diff
@@ -5,23 +5,74 @@
 extends Node
 
 func _ready():
-	var rust_tests := IntegrationTests.new()
-
-	var rust = rust_tests.run()
-	var manual = $ManualFfiTests.run()
-	var generated = $GenFfiTests.run()
-
-	var status: bool = rust && manual && generated
-
+	var test_suites: Array = [
+		IntegrationTests.new(),
+		preload("res://ManualFfiTests.gd").new(),
+		preload("res://gen/GenFfiTests.gd").new(),
+	]
+	
+	var tests: Array[_Test] = []
+	for suite in test_suites:
+		for method in suite.get_method_list():
+			var method_name: String = method.name
+			if method_name.begins_with("test_"):
+				tests.push_back(_Test.new(suite, method_name))
+	
 	print()
-	var exit_code: int
-	if status:
-		print(" All tests PASSED.")
-		exit_code = 0
-	else:
-		print(" Tests FAILED.")
-		exit_code = 1
-
-	print(" -- exiting.")
-	rust_tests.free()
+	print_rich("  [b][color=green]Running[/color][/b] test project %s" % [
+		ProjectSettings.get_setting("application/config/name", ""),
+	])
+	print()
+	
+	var stats := TestStats.new()
+	stats.start_stopwatch()
+	for test in tests:
+		printraw("test %s ... " % [test.test_name])
+		var ok: bool = test.run()
+		print_rich("[color=green]ok[/color]" if ok else "[color=red]FAILED[/color]")
+		stats.add(ok)
+	stats.stop_stopwatch()
+	
+	print()
+	print_rich("test result: %s. %d passed; %d failed; finished in %.2fs" % [
+		"[color=green]ok[/color]" if stats.all_passed() else "[color=red]FAILED[/color]",
+		stats.num_ok,
+		stats.num_failed,
+		stats.runtime_seconds(),
+	])
+	print()
+	
+	for suite in test_suites:
+		suite.free()
+	
+	var exit_code: int = 0 if stats.all_passed() else 1
 	get_tree().quit(exit_code)
+
+class _Test:
```

Comment:
Does it make sense to call this `TestSuite` or so?

Reply:
I see. `TestCase` maybe, but we can also leave it.
Did you add the underscore to make it "private" or to avoid naming conflicts?

Reply:
It's "private": [style guide](https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_styleguide.html#functions-and-variables).